components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -532,8 +542,45 @@ std::vector<mojom::FeedItemV2Ptr> GenerateBlockFromContentGroups(
       },
       std::move(get_weighting));
 
+  base::flat_set<std::string> politics_publisher_ids;
+  for (const auto& [publisher_id, publisher] : publishers) {
+    auto channels = GetChannelsForPublisher(locale, publisher);
+    if (base::Contains(channels, "Politics")) {
```

Comment:
I think we should lift this out of `GenerateBlockFromContentGroups` so we don't recalculate it every block